### DISCUSSION TITLE
Error when using histogram(grains)

### DISCUSSION THREAD

**Discussion by FireFoxBravo** [mtex-toolbox/mtex#1410] [Ask Anything]
I have several EBSD scans of heat treated Titanium-575 samples heat treated at various temperatures. Like most titanium it contains two titanium phases namely alpha and beta titanium, in MTEX labelled as Ti-Hex and Titanium cubic respectively.

I am attempting to produce histograms of relative area (%) against grain area (um^2), as documented in the Grains -> Shape parameters section on the documentation site. When using histogram(grains), I get the following error that prevents it from plotting:

![image](https://user-images.githubusercontent.com/104002913/164987215-e6719946-262e-4ddc-a2d4-f788e89ccc68.png)

The code I have used is shown below:
![image](https://user-images.githubusercontent.com/104002913/164987237-15e8be6a-80a6-413d-9905-17042b4d4374.png)

I can produce the plot using the same code but the older method of hist(grains) as shown below, but I have been advised that the former method is the more "correct one" to use:

![image](https://user-images.githubusercontent.com/104002913/164987272-d975d020-1c9f-4b5a-b543-2ae0b24c6452.png)

![image](https://user-images.githubusercontent.com/104002913/164987338-48bc42ef-19c5-43d7-a4fc-e79408ec5390.png)

I can also plot each phase in isolation using histogram(grains('phase name')); just fine, but this alters the relative area axis, I assume as its only considering the one phase rather than both.

![image](https://user-images.githubusercontent.com/104002913/164987347-f22d3009-044d-43fd-a767-0b03c3c59d72.png)

![image](https://user-images.githubusercontent.com/104002913/164987354-95c4353e-8166-408a-a332-28e4a0ef8e37.png)

Does anyone know what the cause of the error is or can parse out what's its saying , or if I can modify the relative area axis on the individual phase plots such that it correctly reflects the relative area when both phases are considered such as in the hist(grains plot)?

I apologise if my description is not particularly clear, I would be happy to try and clarify.
Thanks.

**Comment by kilir** [comment]
Hi,
as far as I can see, if you upgrade to the latest developer version (or at least apply this commit https://github.com/mtex-toolbox/mtex/commit/0ac004529d77cbf021dd34f78944b437062891d1) this bug should be fixed.

Cheers,
Rüdiger
